### ISSUE TITLE
The program window is wider than the monitor and can't be changed in one-window mode

### ISSUE THREAD

**Issue by neimovirne** [jliljebl/flowblade#1104]
Please include the following information.
1) Flowblade version (Help->About): 2.10
2) MLT version (Help->Runtime Environment): 7.17.0
3) Your distribution (Ubuntu, Debian, Mint etc.): Ubuntu 23.04 GNOME Wayland
4) If possible, try to provide step-by-step instructions on how to reproduce the issue. From developer point of view this is the easieast way to understand what is happening.: Install Flowblade from app store-open it-see that u cant resize app window if it runs in one-window mode

It can be resized only if runs in two-window mode
https://github.com/jliljebl/flowblade/assets/2023595/144343a6-ce1a-4ea1-bddd-456a781f4af4

**Comment by jliljebl** [comment]
Thanks for doing the video, i can see the problem clearly.

A few questions. What is your screen size? Did you try to change the layout from View menu or the button in top row panel, and did that have any effect?

**Comment by jliljebl** [comment]
I have 1920x1080px screen also, so something weird is going on here.

Can see if there is file in your system */home/\<your home folder\>/.config/flowblade/prefs*. If it is there, copy it first somewhere else, then delete it, restart the application, and see if there is any difference?

If that helped, could you upload the *prefs* file here.

**Comment by bitomic** [comment]
Same issue here. There is no `flowblade` folder in `/home/me/.config/`.

**Comment by jliljebl** [comment]
@bitomic are running Wayland? 

The file location is probably */home/you/.var/app/io.github.jliljebl.Flowblade/config/flowblade/prefs*, can you try removing it and see what happens?

Also, are you running a setup with 2 or more monitors?

**Comment by neimovirne** [comment]
> I have 1920x1080px screen also, so something weird is going on here.
> 
> Can see if there is file in your system _/home/<your home folder>/.config/flowblade/prefs_. If it is there, copy it first somewhere else, then delete it, restart the application, and see if there is any difference?
> 
> If that helped, could you upload the _prefs_ file here.

sorry for late answer
tried that, nothing has changed

**Comment by bitomic** [comment]
@jliljebl apologies, I didn't see the notification before. No `.var` (nor `var`) in my home directory. Yes I am using Wayland and with 2 monitors. I can test tomorrow if it makes any difference with a single monitor.

**Comment by jliljebl** [comment]
All additional info helps. I'll test with two monitors next week, but my system doesn't run Wayland currently (maybe because it has NVidia card) so I'll have to see if I can get my older laptop running Wayland to test the issue.

**Comment by jliljebl** [comment]
Couldn't get my previous dual monitor set-up to work, its been a while, and this time second monitor didn't picked up on boot.

Meanwhile any additional confirmation or disproof that problem requires Wayland ? 2 or more monitors setup would be useful, so anyone please post any additional information here if you encounter the problem.

**Comment by neimovirne** [comment]
> Couldn't get my previous dual monitor set-up to work, its been a while, and this time second monitor didn't picked up on boot.
> 
> Meanwhile any additional confirmation or disproof that problem requires Wayland ? 2 or more monitors setup would be useful, so anyone please post any additional information here if you encounter the problem.

I've tried X11 session in Ubuntu, bug still exist

**Comment by jliljebl** [comment]
> I've tried X11 session in Ubuntu, bug still exist

So this is still dual monitor setup? Can you start flatpak with command:

**flatpak run io.github.jliljebl.Flowblade**

and post the part about screen size here, it looks like this on my system with single monitor:

```
Valid shortcut files found: ['flowblade.xml', 'premiere.xml']
Keyboard shortcuts file: flowblade.xml
Gtk version is 3.24.38, Flowblade theme is available.
Using Full Screen size for layout: 1920 x 1080
Small height: False
Small width: False
Detecting environment...
```

I just tried X11 with Gnome 44.2 and dual monitor setup and  and bug did not reproduce here. 

I actually only tried with my dev version, I'll try the Flatpak too, maybe that's the issue.

**Comment by jliljebl** [comment]
No, bug not happening with Flatpak version either. So, not Wayland, not Flatpak it seems. We need to somehow find the difference between systems..

If we can confirm that it is always a dual monitor setup, then we at least have a starting point.

**Comment by neimovirne** [comment]
> > I've tried X11 session in Ubuntu, bug still exist
> 
> So this is still dual monitor setup? Can you start flatpak with command:
> 
> **flatpak run io.github.jliljebl.Flowblade**
> 
> and post the part about screen size here, it looks like this on my system with single monitor:
> 
> ```
> Valid shortcut files found: ['flowblade.xml', 'premiere.xml']
> Keyboard shortcuts file: flowblade.xml
> Gtk version is 3.24.38, Flowblade theme is available.
> Using Full Screen size for layout: 1920 x 1080
> Small height: False
> Small width: False
> Detecting environment...
> ```
> 
> I just tried X11 with Gnome 44.2 and dual monitor setup and and bug did not reproduce here.
> 
> I actually only tried with my dev version, I'll try the Flatpak too, maybe that's the issue.

no, I have single monitor setup, just laptop

**Comment by jliljebl** [comment]
> no, I have single monitor setup, just laptop

Ok, thanks. I'll need to think what other factors could exist, but unfortunately right I don't know where to look.

**Comment by jliljebl** [comment]
Tested on Wayland, didn't reproduce.

**Comment by archerallstars** [comment]
I also have this issue. I have dual monitor setup, GNOME 46 Wayland. 

I'm using the Flatpak version.

> flatpak run io.github.jliljebl.Flowblade
>
> and post the part about screen size here, it looks like this on my system with single monitor:

Mine returns:

```
Valid shortcut files found: ['flowblade.xml', 'premiere.xml']
Keyboard shortcuts file: flowblade.xml
Using Full Screen size for layout: 3072 x 1728
Small height: False
Small width: False
Detecting environment...
```

I suspect that the app doesn't work well with fractional scaling in my system, as both of my monitors' resolution is 1920x1080, but it is detected as 3072 x 1728. I use 125% fractional scaling.

I can't reproduce this issue with fractional scaling disabled at 1920x1080. But since everything is too small for me. So, I try 1600x900 resolution without fractional scaling, the app is still opened larger than my screen, of which I can't resize the app to fit my screen.

I hope this report helps.

**Comment by jliljebl** [comment]
> 1920x1080, but it is detected as 3072 x 1728. I use 125% fractional scaling.

> I can't reproduce this issue with fractional scaling disabled at 1920x1080.

This is definitely a bug, with 2 1920x1080 monitors we should get: 

```
Using Full Screen size for layout: 3072 x 864
Small height: True
Small width: False
Detecting environment...
```

because  *(2 x 1920) / 1.25 = 3072*, and *1080 / 1.25 = 864*. Instead we are getting for vertical dimension *(2 x 1080) / 1.25 = 1728* which is wrong, well, unless monitors are placed on top of each other.

**Comment by jliljebl** [comment]
I found and fixed the bug. I'll  push a hotfix release and comment here once it is available to see if this fixes your issue.

**Comment by archerallstars** [comment]
@jliljebl The issue is still reproducible with dual monitors setup. From the log, now it gets the _correct_ height value: 3072 x _864_. However, it seems the width value is still incorrect.

The app is still open wider than the current monitor width and cannot be resized to fit the monitor width, as shown in the screenshot below:

![s-obfuscated](https://github.com/jliljebl/flowblade/assets/1403194/5849b296-9875-480f-abb6-d42617ee4d9d)

I try with single monitor setup, 1920x1080 @ 125% fractional scaled, the app gets the _correct_ width value (after scaled): _1536_ x 864. The app fits the monitor width well and can be resized to the size smaller than the monitor width, as shown in the screenshot below:

![s2-obfuscated](https://github.com/jliljebl/flowblade/assets/1403194/3fb89e83-f2a9-426c-9a1d-72ee2c5aa381)

**Comment by jliljebl** [comment]
Ok, can you try to following with 2 monitor set-up:

From main menu select *Edit->Preferences* and then in *View* panel select from *Do GUI layout based on:*  some other option then *Full Display Area 3072 x 864*. It should show both monitors as options. Now things should work, right?

**Comment by archerallstars** [comment]
@jliljebl Yes, this fixed the issue. I changed to **Monitor 0: 1536 x 864**.
